### PR TITLE
refactor(LocalStorage): FRMW-636-convert LocalStorage to Quark.

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-636

- [x] Dev LGTM
- [ ] Dev LGTM
- [x] Design LGTM

---
Refactor `rxLocalStorage.LocalStorage` to `rxLocalStorage` Quark Service.  Deprecating `LocalStorage` service.

### CODE REVIEW

**Comment by 100stacks** [comment]
## `LocalStorage` Quark Service

### Demo Page
screen shot 2016-01-12 at 2 34 07 pm

### API Docs
screen shot 2016-01-12 at 2 35 12 pm

**Comment by 100stacks** [comment]
**TODO:** Add `{@link quarks.service:LocalStorage LocalStorage}` to `quarks.module.js`.

File: src/components/rxPaginate/rxPaginate.module.js [hunk 1]
```diff
@@ -57,7 +57,7 @@
  * ## Persistence
  *
  * The user's preference for the number of items to display per page will be persisted across applications
- * using {@link rxLocalStorage}. This preference is set whenever the user selects a new number to show.
+ * using LocalStorage. This preference is set whenever the user selects a new number to show.
```
Comment (CITguy) [inline src/components/rxPaginate/rxPaginate.module.js]: Please link to the new jsdoc location

File: src/quarks/LocalStorage/docs/LocalStorage.midway.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+var LocalStorage = encore.LocalStorage;
```
Comment (CITguy) [inline src/quarks/LocalStorage/docs/LocalStorage.midway.js]: Please place this variable definition within the outer `describe` block below to reduce global var pollution.

File: src/quarks/LocalStorage/scripts/LocalStorage.page.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 var Page = require('astrolabe').Page;
 
-exports.rxLocalStorage = Page.create({
```
Comment (CITguy) [inline src/quarks/LocalStorage/scripts/LocalStorage.page.js]: Please create an alias for this definition below the export statement.

```javascript
exports.LocalStorage = Page.create({ ... });

/*
 * @deprecated
 * @exports encore.rxLocalStorage
 * @description
 * **DEPRECATED**: Please use `encore.LocalStorage` as a stand-in replacement.
 */
exports.rxLocalStorage = exports.LocalStorage;
```

**Comment by glynnis** [comment]
Design LGTM

**Comment by 100stacks** [comment]
@CITguy, comments implemented...great job of tracking down the deep dependency issue :tada:

File: src/quarks/quarks.module.js [hunk 4]
```diff
@@ -30,9 +30,10 @@
  * * {@link quarks.service:rxSortUtil rxSortUtil}
  * * {@link quarks.service:Identity Identity}
  * * {@link quarks.service:rxStatusMappings rxStatusMappings}
- * * {$link quarks.service:ErrorFormatter ErrorFormatter}
```
Comment (100stacks) [inline src/quarks/quarks.module.js]: Fixing typo

File: src/quarks/LocalStorage/docs/examples/LocalStorage.simple.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<div ng-controller="LocalStorageSimpleCtrl">
+  <p>Select <strong>Store Answer</strong>, then <code>Answer?</code> to first store the answer in the
+     browser's <code>localStorage</code> object and later retrieve the stored content.
+  </p>
```
Comment (CITguy) [inline src/quarks/LocalStorage/docs/examples/LocalStorage.simple.html]: Please place the descriptive text in `src/quarks/LocalStorage/docs/LocalStorage.docs.html` to keep the example markup as small and focused as possible.

**Comment by 100stacks** [comment]
@CITguy, changes made.  Also updated the Demo Page screenshot in https://github.com/rackerlabs/encore-ui/pull/1494#issuecomment-171061352. :+1:

File: src/quarks/LocalStorage/docs/LocalStorage.docs.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<p>
+  Simple wrapper for interacting with local storage in the browser.
+</p>
+<p>Select <code>Store Answer</code>, then <code>Answer?</code> to first store the answer in the
```
Comment (CITguy) [inline src/quarks/LocalStorage/docs/LocalStorage.docs.html]: Please use an appropriate heading between the first and second paragraphs to better organize the document. I'd go with something like `<h3>Simple Example</h3>`

**Comment by CITguy** [comment]
@100stacks: considering that we're already in the codebase for this service. I'd like to rename the service to `rxLocalStorage` (instead of the current `LocalStorage`). My reasoning behind this is that the service name `LocalStorage` (UpperCamelCase) is too similar to the native API name of `localStorage` (lowerCamelCase). As such, please 

* change service name to `rxLocalStorage`
* revert deprecation/alias in page object
* create alias for `LocalStorage` to retain backward compatibility (See below)

### Alias `LocalStorage`
Define in `src/quarks/rxLocalStorage/scripts/LocalStorage.js`.
```javascript
angular.module('encore.ui.quarks')
/**
 * @ngdoc service
 * @deprecated
 * @description
 * **NOTICE:** This service has been deprecated in favor of the 
 * {@link quarks:service.rxLocalStorage rxLocalStorage} service.
 */
.service('LocalStorage', function (rxLocalStorage) {
  return rxLocalStorage;
});
```

File: src/components/rxPaginate/rxPaginate.module.js [hunk 7]
```diff
@@ -57,7 +57,8 @@
  * ## Persistence
  *
  * The user's preference for the number of items to display per page will be persisted across applications
- * using {@link rxLocalStorage}. This preference is set whenever the user selects a new number to show.
+ * using {@link quarks.service:LocalStorage LocalStorage}. This preference is set whenever the user selects
```
Comment (CITguy) [inline src/components/rxPaginate/rxPaginate.module.js]: Update link to point to rxLocalStorage service.

File: src/components/rxSession/scripts/Session.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ angular.module('encore.ui.rxSession')
  *
  * Service for managing user session in encore-ui.
  *
- * @requires rxLocalStorage.service:LocalStorage
+ * @requires quarks.service:LocalStorage
```
Comment (CITguy) [inline src/components/rxSession/scripts/Session.js]: Update to point to `quarks.service:rxLocalStorage`

File: src/quarks/LocalStorage/scripts/LocalStorage.spec.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ describe('service:LocalStorage', function () {
     var storage;
 
     beforeEach(function () {
-        module('encore.ui.rxLocalStorage');
+        module('encore.ui.quarks');
         inject(function (LocalStorage) {
```
Comment (CITguy) [inline src/quarks/LocalStorage/scripts/LocalStorage.spec.js]: Change to `rxLocalStorage`

**Comment by 100stacks** [comment]
@CITguy, ok pushed updates to `rxLocalStorage` Quark Service, and deprecating `LocalStorage` Quark Service. 

Updated screenshots in https://github.com/rackerlabs/encore-ui/pull/1494#issuecomment-171061352

File: src/components/rxFeedback/rxFeedback.module.js [hunk 10]
```diff
@@ -126,9 +126,6 @@
  *
  * ## Directives
  * * {@link rxFeedback.directive:rxFeedback rxFeedback}
- *
```
Comment (100stacks) [inline src/components/rxFeedback/rxFeedback.module.js]: Maintenance from Quark Shuffle.

Comment (CITguy) [inline src/components/rxFeedback/rxFeedback.module.js]: Thank you

File: src/quarks/quarks.module.js [hunk 11]
```diff
@@ -30,9 +30,8 @@
...
- * * {$link quarks.service:rxFeedbackSvc rxFeedbackSvc}
```
Comment (CITguy) [inline src/quarks/quarks.module.js]: Don't forget this link. I see it's missing in the update.

File: src/quarks/rxLocalStorage/scripts/rxLocalStorage.spec.js [hunk 12]
```diff
@@ -14,51 +14,51 @@ describe('service:LocalStorage', function () {
         storage.clear();
     });
 
-    it('LocalStorage should exist', function () {
+    it('rxLocalStorage should exist', function () {
```
Comment (CITguy) [inline src/quarks/rxLocalStorage/scripts/rxLocalStorage.spec.js]: All of these `it()` descriptions that you've updated can be changed to remove `rxLocalStorage` from the description. The subject being repeated in each of the `it()` descriptions is already determined by the overall `describe()` description. The `it` functions should read like a sentence fragment, but currently they are a bit wordy and nonsensical. 

For example, this test description can be updated to...

```javascript
it('should exist', function () { /* ... */ })
```

**Comment by 100stacks** [comment]
@CITguy, made the ^^^ updates. :+1:

**Comment by CITguy** [comment]
LGTM

**Comment by parlarjb** [comment]
LGTM

File: src/quarks/quarks.module.js [hunk 13]
```diff
@@ -30,9 +30,9 @@
...
- * * {$link quarks.service:rxNotify rxNotify}
```
Comment (CITguy) [inline src/quarks/quarks.module.js]: Note: rxNotify hasn't been pulled into Quarks as of this PR.
